yew-macro/src/html_tree/html_tag/tag_attributes.rs

```diff
@@ -307,14 +307,59 @@ impl Parse for TagAttributes {
             i += 1;
         }
         let booleans = TagAttributes::drain_boolean(&mut attributes);
+        for boolean in &booleans {
+            if boolean.question_mark.is_some() {
+                return Err(syn::Error::new_spanned(
+                    &boolean.label,
+                    format!(
+                        "The '{}' attribute does not support being used as an optional attribute",
+                        boolean.label
+                    ),
+                ));
+            }
+        }
 
-        let classes =
-            TagAttributes::remove_attr(&mut attributes, "class").map(TagAttributes::map_classes);
+        let classes = TagAttributes::remove_attr(&mut attributes, "class");
+        if let Some(classes) = classes.as_ref() {
+            if classes.question_mark.is_some() {
+                return Err(syn::Error::new_spanned(
+                    &classes.label,
+                    "The 'class' attribute does not support being used as an optional attribute",
+                ));
+            }
+        }
+        let classes = classes.map(|a| TagAttributes::map_classes(a.value));
         let value = TagAttributes::remove_attr(&mut attributes, "value");
         let kind = TagAttributes::remove_attr(&mut attributes, "type");
         let checked = TagAttributes::remove_attr(&mut attributes, "checked");
+        if let Some(checked) = checked.as_ref() {
+            if checked.question_mark.is_some() {
+                return Err(syn::Error::new_spanned(
+                    &checked.label,
+                    "The 'checked' attribute does not support being used as an optional attribute",
+                ));
+            }
+        }
         let node_ref = TagAttributes::remove_attr(&mut attributes, "ref");
+        if let Some(node_ref) = node_ref.as_ref() {
+            if node_ref.question_mark.is_some() {
+                return Err(syn::Error::new_spanned(
+                    &node_ref.label,
+                    "The 'ref' attribute does not support being used as an optional attribute",
+                ));
+            }
+        }
+        let node_ref = node_ref.map(|n| n.value);
         let key = TagAttributes::remove_attr(&mut attributes, "key");
+        if let Some(key) = key.as_ref() {
```

Comment:
```suggestion
        if let Some(ref key) = key {
```

Reply:
Some reason you prefer that one over the other? The [atch ergonomics RFC](https://github.com/rust-lang/rfcs/blob/master/text/2005-match-ergonomics.md) suggests that
> The `ref` keyword is a pain for Rust beginners, and a bit of a wart for everyone else. It violates the rule of patterns matching declarations, it is not found anywhere outside of patterns, and it is often confused with `&`.

Reply:
This is really just preference and I shouldn't have added these suggestions in hindsight.

For me it's that I prefer using dedicated syntax over a method call.
A much less controversial example would be writing `a == b` instead of  `a.eq(&b)`.

I marked the other instances as "resolved" but I'm leaving this one for visibility.

Reply:
If you want, I can change `x.as_ref()` to `&x`, which I believe should do the same thing as your change without invoking `ref`.

Reply:
That sounds good